aristanetworks/ansible-avd
# Feat(eos_cli_config_gen)!: Added support for snmp v1 & snmp v2c

## Change Summary

Implement changes discussed in issue 1033.

## Types of changes


## Related Issue(s)

Fixes #1033 

## Component(s) name

`arista.avd.roles.eos_cli_config_gen.templates.eos.snmp-settings.j2`

## Proposed changes
Added support for SNMPv1 and SNMPv2c trap receivers by updating the schema formerly found in snmp_v3.yml, moving to snmp.yml, and updating snmp-settings.j2.  The primary change was to deal with the schema modification which put the version key under the host instead of under the username where it was only relevant for SNMP v3.  

Old Schema:
 
```yaml
 hosts:
    - host: <host>
      vrf: <vrf>
      users:
        - username: <username>
          authentication_level: < auth | noauth | priv >
          version: < 1 | 2c | 3 >
```

New Schema:

```yaml
  hosts:
    - host: <host>
      vrf: <vrf>
      version: < 1 | 2c | 3 >
      communities:
        community: <community>
      users:
        - username: <username>
          authentication_level: < auth | noauth | priv >
```

## How to test
Added SNMP intent data in YML to group_vars, manually ran playbook to generate configs and validated syntax was as expected and manually validated against EOS switch running 4.26.1.  Additionally verified that  arista/avd/molecule/eos_cli_config_gen/intended/configs/snmp.cfg was appropriately generated after running make converge.

## Checklist:


- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
